enhancements/samples-generation.md

```diff
@@ -0,0 +1,613 @@
+
+The e2e test also will be able to invoke the samples code implementation which will ensure that they are runnable and will keep centralized this implementation as well to achieve a better maintainability.
+
+In this way, it is no longer required keep the repository [operator-sdk-samples](https://github.com/operator-framework/operator-sdk-samples) and it should be deprecated. 
+
+**NOTE** The test utils as what is proposed here can fit well in upstream. It is not a Goal push these implementations to upstream, however, it definitely could be an ultimate end goal to keep in mind. 
+
+## Motivation
+
+### Goals
+
+- Generate samples via Go to improve the maintainability, readability and reusability
+- Checking if samples are outdated via CI to reduce its risk of the sample projects gets outdated
+- Building samples on SDK to facilitate the development process to mock data and keep the samples maintained by using the same approach and design defined and used via the test
+- Re-using samples on tests to improve the maintainability and ensure that they are runnable.
+- Helpers and utilities available for tests and samples
+- Specific implementation following the same approach to define the sample and the tests context
+
+### Non-Goals
+
+- Link the samples in the docs or aligned then with the tutorials
+- Provide advanced helpers or Boilerplates to mock the data that could be used in the tests and/or in the samples generation
+- Push this implementation to upstream
+
+## Proposal
+
+### User Stories
+
+#### Generate samples via Go 
+
+- I am a maintainer/contributor, I want to be able to troubleshoot the code used to create the samples, so that I can easily fix the issues. 
+- I am a maintainer/contributor, I want to be develop the samples as I develop the tests, so that I do not need to be familiar with a different approach
+- I am a maintainer/contributor, I want to be easily create samples based on the tutorial steps, so that we can provide the expected result of the website tutorials
+
+#### Checking if samples are outdated via CI
+
+- I am a maintainer/contributor, I want to be able to check the samples update with the changes made, so that I do not need to spend effort on the samples after the release process
+- I am a maintainer/contributor, I want to able to check the samples updated in the diff of te github on the review process, so that I do not need to clone them locally
+
+#### Building samples on SDK
+
+- I am a maintainer/contributor, I want to be easily ensure that the changes made will not brake the steps required to build the projects, so that we can reduce the chance to introduce critical bugs. 
+- I am a maintainer/contributor, I want to check the result of the changes made in the scaffold by verifying the samples changes , so that I do not need to pull the pull request locally and run the steps to review the code. 
```

Comment:
```suggestion
- I am a maintainer/contributor, I want to check the result of the changes made in the scaffolding by verifying the samples changes , so that I do not need to pull the pull request locally and run the steps to review the code. 
```